src/plugins/inspector/public/views/requests/components/details/req_details_stats.tsx

```diff
@@ -28,58 +28,62 @@ import {
   EuiTableRowCell,
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import { Request, RequestStatistic } from '../../../../adapters/request/types';
+import { RequestDetailsProps } from '../types';
 
-class RequestDetailsStats extends Component {
-  static shouldShow = (request) => !!request.stats && Object.keys(request.stats).length;
+// TODO: Replace by property once available
+interface RequestDetailsStatRow extends RequestStatistic {
+  id: string;
+}
+
+export class RequestDetailsStats extends Component<RequestDetailsProps> {
+  static propTypes = {
+    request: PropTypes.object.isRequired,
```

Comment:
nit: Since `RequestDetailsProps` is used for a few components and only consists of the `Request` interface, I wonder if we should instead move the props interface to each of the components themselves, and then import the `Request` interface there instead?

Maybe I'm just used to the convention of colocating props interfaces next to the components they belong to.

Also, do we get any additional type safety from adding `propTypes` to components that already have their props defined as TS interfaces? We might be able to remove `propTypes` from these components entirely...

Reply:
about 1 question, not sure that we can do it, unfortunately we have a ts rule which doesn't allow to add empty interface: 
![image](https://user-images.githubusercontent.com/20072247/63160190-de745680-c025-11e9-9960-6e3f25773761.png)

as for  `propTypes`, I like the idea of using it because it allows us to find errors at runtime.